tsk/fs/hfs.c

```diff
@@ -1730,6 +1730,15 @@ hfs_cat_file_lookup(HFS_INFO * hfs, TSK_INUM_T inum, HFS_ENTRY * entry,
 }
 
 
+static uint8_t
+hfs_find_highest_inum_cb(HFS_INFO * hfs, int8_t level_type,
+    const hfs_btree_key_cat * cur_key,
+    TSK_OFF_T key_off, void *ptr)
+{
+  *((TSK_INUM_T*) ptr) = tsk_getu32(hfs->fs_info.endian, cur_key->parent_cnid);
```

Comment:
I'm fine with adding that check, but I don't understand why it would be necessary. Mustn't the callback be called in tree traversal order? How else would the traversal function know which branch to take?

And if the ordering constraints of the tree are violated, I'd think you'd have a lot bigger problems than the wrong highest inum: you wouldn't be able to find anything.